### ISSUE TITLE
strange error messages for missing right parenthesis (or curly brace)

### ISSUE THREAD

**Issue by corpusc** [skycoin/cx#393]
**Describe the bug**

although this is a bottom priority thing, and it is fairly obvious what the problem is in my example code,... when you have a long/complex line/expression, it can take awhile to figure out what the problem is, when the error refers to characters that may not be present in the line/expression.  which is what happened to me.  :)  and CX code is heavy with parentheses pairs with all the longhand expressions currently needed (for reliability).

**To Reproduce**

1. use the following code:

package main

var sl []i32 = []i32{1,2,3}

func main () {
	printf("len: %d", len(sl)
}

2. now delete the the right curly brace from line 3, for the same error message (pointing to different line).

**Expected behavior**

i would like the error to mention that a matching right parenthesis (or curly brace) is needed.

**Desktop**
 - OS:  Windows 10
 - CX Version 0.7beta

**Comment by ingwal** [comment]
Yes, the error messages of CX are not good.  We are working on this continuously, but the progress is not as fast as we would wish.  I will see what we can do for this particular case, since it is rather common.
